wallet/src/main/scala/org/constellation/wallet/Utils.scala

```diff
@@ -120,7 +134,15 @@ case class TypedEdgeHash(hash: String, hashType: EdgeHashType, baseHash: Option[
 case class ObservationEdge(
   parents: Seq[TypedEdgeHash],
   data: TypedEdgeHash
-)
+){
+  def hash = Hashing.sha256().hashBytes(KryoSerializer.serializeAnyRef(getRunLengthEncoding)).toString
+  def getRunLengthEncoding = {
```

Comment:
`getRunLengthEncoding` should be abstracted somehow. Either in a separate trait or ideally I'd see an adapter for the hardware wallet which will be able to convert our structure into the hardware wallet compatible one.

Also, the "frame" should be tested and typed if possible. Now, you're able to return everything, in various order, which is very error prone.

Reply:
We should probably add it to Signable. We can replace hash there too

Reply:
Yes, you can put a function declaration into a `Signable` and use it in a hash fn. It should be fine for now ;)